### PR TITLE
Ignore Configmap mounted in DeploymentConfig

### PR DESCRIPTION
**Pull request by mcouliba**
**What type of PR is this?**
/kind bug

**What does does this PR do / why we need it**:
When creating applications on OpenShift Kubernetes, a (12-factor) best practice is to externalize the configuration in a ConfigMap. When mounting this ConfigMap in the DC, odo is not working anymore because it expects a volume with a PVC. As odo cannot manage ConfigMap yet, we should ignore them (as we ignore empty volume) to  apply best practices while using odo.

**Which issue(s) this PR fixes**:

This is an alternance for waiting for the [PR2142](https://github.com/openshift/odo/issues/2142) and [PR2601](https://github.com/openshift/odo/issues/2601)

**PR acceptance criteria**:

- [ ] Unit test 

- [ ] Integration test 

- [ ] Documentation 

- [ ] I have read the [test guidelines](https://github.com/openshift/odo/blob/master/docs/dev/test-architecture.adoc)

**How to test changes / Special notes to the reviewer**:

### CODE REVIEW

File: pkg/storage/storage.go [hunk 1]
```diff
@@ -147,6 +147,11 @@ func List(client *occlient.Client, componentName string, applicationName string)
 			continue
 		}
 
+		// We should ignore ConfigMap (while PR2142 and PR2601 are not fixed)
+		if client.IsVolumeAnConfigMap(volumeMount.Name, dc) {
```
Comment (mik-dass) [inline pkg/storage/storage.go:151]: The client currently has no such function. Thus the build is failing
```
pkg/storage/storage.go:151:12: client.IsVolumeAnConfigMap undefined (type *occlient.Client has no field or method IsVolumeAnConfigMap)
```

**Comment by mik-dass** [comment]
> How to test changes / Special notes to the reviewer:

Can you please provide some steps to test these changes?

File: pkg/storage/storage.go (hunk 1 above)
Comment (mcouliba) [reply pkg/storage/storage.go:151]: Yes sorry! Just committed it

**Comment by mcouliba** [comment]
To reproduce the error, please follow:

- Create a component using odo (`odo create java:11 my-component --app my-app`)
- Create a ConfigMap (`oc create configmap cm-test`)
- Mount the ConfigMap in the DC created by odo (`oc set volume dc/my-component --add --configmap-name=cm-test --mount-path=/tmp`)
- Run the command `odo list --app my-app`

Without the fix, you will have the message 'no PVC associated with Volume Mount volume-xxxx'

**Comment by mcouliba** [comment]
@mik-dass The step "make test-cmd-push" failed but where can I get more logs about the error.

**Comment by mik-dass** [comment]
> @mik-dass The step "make test-cmd-push" failed but where can I get more logs about the error.

The odo verbose error log is here https://travis-ci.com/github/openshift/odo/jobs/428010783#L1147

File: pkg/occlient/occlient.go [hunk 2]
```diff
@@ -3069,6 +3069,18 @@ func (c *Client) IsVolumeAnEmptyDir(volumeMountName string, dc *appsv1.Deploymen
 	return false
 }
 
+// IsVolumeAnEmptyDir returns true if the volume is an EmptyDir, false if not
```
Comment (mik-dass) [inline pkg/occlient/occlient.go:3072]: Can you please update this comment? Rest seems to work for me.

Comment (mcouliba) [reply pkg/occlient/occlient.go]: Done.

**Review by mik-dass — APPROVED** [review]
/lgtm

**Comment by mcouliba** [comment]
hello @mik-dass, I have spent some time trying to fix the Travis CI. It is very weird because each time I run a test (`make test-xxx`) I've got different problems and some tests OK became KO and viceversa. 

Is it a expected behaviour? If so, how can I fix my PR to make it all green?

**Comment by mik-dass** [comment]
> hello @mik-dass, I have spent some time trying to fix the Travis CI. It is very weird because each time I run a test (`make test-xxx`) I've got different problems and some tests OK became KO and viceversa.
> 
> Is it a expected behaviour? If so, how can I fix my PR to make it all green?

The tests are failing due a flake https://github.com/openshift/odo/issues/3454. You can restart the jobs to make Travis go green.

File: pkg/occlient/occlient.go [hunk 3]
```diff
@@ -3069,6 +3069,18 @@ func (c *Client) IsVolumeAnEmptyDir(volumeMountName string, dc *appsv1.Deploymen
 	return false
 }
 
+// IsVolumeAnConfigMap returns true if the volume is an ConfigMap, false if not
+func (c *Client) IsVolumeAnConfigMap(volumeMountName string, dc *appsv1.DeploymentConfig) bool {
+	for _, volume := range dc.Spec.Template.Spec.Volumes {
+		if volume.Name == volumeMountName {
+			if volume.ConfigMap != nil {
+				return true
```
Comment (dharmit) [inline pkg/occlient/occlient.go:3077]: Wouldn't `if volume.Name == volumeMountName && volume.ConfigMap != nil` be better? My only reasoning for saying it "better" is that it is a one-liner and reduces a nested `if`. :)

**Review by dharmit — COMMENTED** [review]
Don't have a strong opinion on the question/suggestion I made so approving.

/approve

**Pull request merged**